doc/rst/platforms/cray.rst

```diff
@@ -201,7 +201,29 @@ Using Chapel on a Cray System
         source util/setchplenv.bash
 
 
-2) Compile your Chapel program.  For example:
+2) On HPE Cray EX systems with ``CHPL_COMM=ofi``, optionally, load the
+   Cray PMI modules::
+
+      module load cray-pmi{,-lib}
```

Comment:
Are both modules needed? When I was testing gasnet I think I only needed `cray-pmi`

Reply:
I think comm=ofi may react differently to these modules being present or not than comm=gasnet does. In particular, I'd guess that GASNet has its own allgather and broadcast support (if it needs that at all) and doesn't have to rely on PMI for those, either though the KVS stuff or the Cray-specific functions. The comm=ofi PMI-based out-of-band support does need one or the other of those, however. Nevertheless, I've taken a more formal look at this now. I built with neither module loaded, with just `cray-pmi` loaded, and both that and `cray-pmi-lib` loaded. Then I ran all three of those executables with `PMI_MAX_KVS_ENTRIES` forced to a value too small to support the comm=ofi out-of-band support, with neither module loaded, just `cray-pmi`, and both loaded. Here are the results:

| | Built with neither | Built with only `cray-pmi` | Built with both |
| --- | --- | --- | --- |
| Run with neither | KVS err | KVS err | ran fine |
| Run with only `cray-pmi` | KVS err | KVS err | ran fine |
| Run with both | KVS err | KVS err | ran fine |

So it seems what's loaded when you build has an effect, but not what's loaded when you run. I will slightly adjust the text to reflect this.

(For completeness, I also started to try out the "only `cray-pmi-lib`" case, but I got link-time errors building the program. This seemed familiar; I think when I was first looking at this stuff several months ago I found there was a dependence from the `cray-pmi-lib` module back on the `cray-pmi` module. I didn't pursue this further.)